conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/processing/AccountsProcessing.scala

```diff
@@ -0,0 +1,301 @@
+package tech.cryptonomic.conseil.indexer.tezos.processing
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
+import akka.Done
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.Source
+import tech.cryptonomic.conseil.indexer.config.{BakingAndEndorsingRights, BatchFetchConfiguration}
+import tech.cryptonomic.conseil.indexer.tezos.{TezosNodeOperator, TezosDatabaseOperations => TezosDb}
+import tech.cryptonomic.conseil.indexer.tezos.TezosNodeOperator.LazyPages
+import tech.cryptonomic.conseil.common.tezos.Tables
+import tech.cryptonomic.conseil.common.tezos.TezosTypes.{
+  Account,
+  AccountId,
+  BlockHash,
+  BlockReference,
+  BlockTagged,
+  Protocol4Delegate,
+  PublicKeyHash,
+  Voting
+}
+import tech.cryptonomic.conseil.indexer.tezos.TezosErrors.AccountsProcessingFailed
+import tech.cryptonomic.conseil.common.tezos.TezosTypes.Syntax._
+
+import com.typesafe.scalalogging.LazyLogging
+import slick.jdbc.PostgresProfile.api._
+
+class AccountsProcessing(
+    nodeOperator: TezosNodeOperator,
+    db: Database,
+    batchingConf: BatchFetchConfiguration,
+    rightsConf: BakingAndEndorsingRights
+)(
+    implicit mat: ActorMaterializer
+) extends LazyLogging {
+
+  type AccountsIndex = Map[AccountId, Account]
+  type DelegateKeys = List[PublicKeyHash]
+
+  /* Fetches the data from the chain node and stores accounts into the data store.
+   * @param ids a pre-filtered map of account ids with latest block referring to them
+   * @param onlyProcessLatest verify that no recent update was made to the account before processing each id
+   *        (default = false)
+   */
+  private def process(
+      ids: Map[AccountId, BlockReference],
+      votingData: Map[BlockHash, List[Voting.BakerRolls]],
+      onlyProcessLatest: Boolean = false
+  )(
+      implicit ec: ExecutionContext
+  ): Future[List[BlockTagged[DelegateKeys]]] = {
+    import cats.Monoid
+    import cats.instances.future._
+    import cats.instances.int._
+    import cats.instances.list._
+    import cats.instances.option._
+    import cats.instances.tuple._
+    import cats.syntax.flatMap._
+    import cats.syntax.functorFilter._
+    import cats.syntax.monoid._
+
+    def logWriteFailure: PartialFunction[Try[_], Unit] = {
+      case Failure(e) =>
+        logger.error("Could not write accounts to the database")
+    }
+
+    def logOutcome: PartialFunction[Try[(Option[Int], Option[Int], _)], Unit] = {
+      case Success((accountsRows, delegateCheckpointRows, _)) =>
+        logger.info(
+          "{} accounts were touched on the database. Checkpoint stored for{} bakers.",
+          accountsRows.fold("The")(String.valueOf),
```

Comment:
Are you sure, that whitespace is not needed here?

Reply:
Look at the previous reply